tests/unit/data-access/FetchFormatValueRepository.spec.ts

```diff
@@ -0,0 +1,73 @@
+import FetchFormatValueRepository from '@/data-access/FetchFormatValueRepository';
+import TechnicalProblem from '@/data-access/errors/TechnicalProblem';
+import ParseResult from '@/data-access/ParseResult';
+
+const testLang = 'en';
+const testEndpoint = 'https://example.com/w/api.php';
+const propertyId = 'P577';
+const valueData: ParseResult = {
+	value: {
+		time: '+1920-02-00T00:00:00Z',
+		timezone: 0,
+		before: 0,
+		after: 0,
+		precision: 10,
+		calendarmodel: 'http://www.wikidata.org/entity/Q1985727',
+	},
+	type: 'time',
+};
+
+describe( 'FetchFormatValueRepository', () => {
+
+	it( 'format date from valueData returned from ParseValueRepository', async () => {
+		const repo = new FetchFormatValueRepository(
+			testEndpoint,
+			testLang,
+		);
+		const expectedResult = 'February 1920';
+
+		window.fetch = jest.fn().mockImplementation( () => Promise.resolve( {
+			ok: true,
+			json: async () => ( expectedResult ),
```

Comment:
What is returned by the API is not `February 1920`, but `{"result":"February 1920"}`

Reply:
fixed

Reply:
Your _expected result_ is not `{"result":"February 1920"}`, but `"February 1920"`. You need to differentiate between what the API returns and what you want your repository to return.

Currently, the repository is actually directly returning the object it receives from the API.